### DISCUSSION TITLE
HNSW parameter limits

### DISCUSSION THREAD

**Discussion by ramialt** [milvus-io/milvus#24873] [Q&A and General discussion] [answered]
Hello,
I was wondering where the HNSW parameter limits from the [documentation](https://milvus.io/docs/index.md) come from?
In the hnsw library Milvus's hnsw algorithm is using, [there are no such limits to the hnsw parameters](https://github.com/nmslib/hnswlib/blob/master/ALGO_PARAMS.md). 

From the [change of a previous issue](https://github.com/milvus-io/milvus/issues/2637) about this, it seems that increasing this limit is just a matter of increasing a couple of constants? I was looking at increasing the range for efConstruction, but I wonder: why do those constant limits exist?

**Comment by yhmo** [comment, marked as answer]
We have changed(less limits) the HNSW parameter ranges in recent versions. Will update the doc once confirmed.

**Reply by ramialt** [reply]
Happy to hear that! [Is this the correct file?](https://github.com/milvus-io/knowhere/blob/main/src/index/hnsw/hnsw_config.h) Looks like the range has been changed to [1, max_int] for ef & ef_construction, and [1, 2048] for M? Is that right?

**Reply by yhmo** [reply]
Right
